parlai/tasks/cornell_movie/agents.py

```diff
@@ -63,20 +58,25 @@ def rebuild(entries):
                         for i in range(len(entries) - 1)
                     ]
                 )
-                if len(entries[-1]) > 1 and entries[-1][1]:
-                    # add y_n => '', if last y avail
-                    new_list.append((entries[-1][1][0], None))
             return new_list
 
+        def is_valid(entry):
+            if entry[0] == '' or entry[1] is None:
+                return False
+            return True
+
         # this shows conversations in both directions
         alternate = []
         for entry, new in super().setup_data(path):
             if new:
                 for i, e in enumerate(rebuild(alternate)):
-                    yield e, i == 0
+                    if is_valid(e):
+                        yield e, i == 0
```

Comment:
Does this create a problem where the first entry is skipped and therefore you end up with two episodes running together?

Reply:
Great, thanks for explaining. And the main teacher seems fine still?

Reply:
Not really. The DefaultTeacher has the same problems, except that empty source examples `['', y1]` only come from training data. I'll make a fix for it later.

Reply:
Well one of our group has been using it lately and so I'm surprised they're not having a problem. You can submit a patch but it might have to wait.

Reply:
Sure. The problem of `None` target only appears occasionally, when a batch is full of such examples. I met that only on the test set, and I assume simple methods like using a different batch size can evade this problem. Anyways, we can leave this for future pull requests.